Bluejay/Bluejay/SynchronizedPeripheral.swift

```diff
@@ -134,11 +146,21 @@ public class SynchronizedPeripheral {
             })
         }
         
-        _ = sem.wait(timeout: DispatchTime.distantFuture)
+        _ = sem.wait(timeout: timeoutInSeconds == 0 ? .distantFuture : .now() + .seconds(timeoutInSeconds))
         
         if let error = error {
+            backupTermination = nil
             throw error
         }
+        else if listenResult == nil {
+            backupTermination = nil
+            
+            if self.parent.isListening(to: characteristicIdentifier) && self.bluetoothAvailable {
+                self.parent.endListen(to: characteristicIdentifier)
```

Comment:
This business of ending the listen only if bluetooth is available makes me a little nervous (it exists other places, so I'm not sure if it's worth fixing just here, I think I'll file a bug to look at that in general), it seems like this could lead to unintentional double listens. I feel like we probably should be clearing out the listen from the table in these cases, and just accepting that notifications might stay turned on on the device in this case.

Reply:
Actually, I think I just realized what must be going on here, if bluetooth isn't available, the device is probably just about to be disconnected, so the fact that we don't clear the listens PROBABLY doesn't matter, because the device and its listen table is just about to be destroyed anyway. So maybe it doesn't actually matter in this case.

Reply:
I believe that is correct. The peripheral object containing the listen table should be destroyed at the end of bluetooth becoming unavailable.